CUDA does not compile

When trying to compile with CUDA, I consistently get errors like

[  1%] Building NVCC (Device) object src/core/CMakeFiles/EspressoCuda.dir/EspressoCuda_generated_p3m_gpu_error_cuda.cu.o
nvcc fatal   : Unsupported gpu architecture 'compute_52'
CMake Error at EspressoCuda_generated_p3m_gpu_error_cuda.cu.o.cmake:203 (message):
  Error generating
  /home/btpj/bt301429/progs/espressoPy/buildCUDA/src/core/CMakeFiles/EspressoCuda.dir//./EspressoCuda_generated_p3m_gpu_error_cuda.cu.o

The file where the error is reported changes (depending on the order in which they are compiled by make -j), it seems to happen with all .cu files.
I tried CUDA 6.5 and 8.0, same result.
On the very same system, ESPResSo 3.3.1 runs perfectly with CUDA.

Any suggestions are welcome! Thanks

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,7 +100,7 @@ endforeach(func_name __PRETTY_FUNCTION__ __FUNCTION__)
 ######################################################################
 
 if (WITH_CUDA)
-  find_package(CUDA QUIET)
+  find_package(CUDA 7.0)
   if(CUDA_FOUND)
     list(APPEND LIBRARIES ${CUDA_CUFFT_LIBRARIES})
     list(APPEND LIBRARIES ${CUDA_LIBRARIES})

